[Synthetics] Group by config !!

shahzad31 opened the pull request:
## Summary

Reworks the Synthetics overview page (with `Group by config` as the foundation) and the monitor detail flyout to make per-location triage faster and the data more reliable. Highlights:

- New **Group by config** mode for the overview, so a single card/row represents one monitor across all of its locations.
- Compact (table) view: row click now reliably opens the detail flyout from anywhere in a row, and a loading state replaces the bare empty placeholder during initial load.
- Detail flyout: redesigned per-location selector, embedded status heatmap, duration chart with an "All locations" toggle, an "Edit monitor" action, and a wider/resizable flyout.
- Histogram fix: project monitors no longer render empty downtime histograms.

### Overview page

Table view

<img width="1728" alt="image" src="https://github.com/user-attachments/assets/073130d4-b3f2-4aa1-baba-480a0883b566" />

Card view

<img width="1728" alt="image" src="https://github.com/user-attachments/assets/7e8fb9bd-bcbd-435e-8bed-cdfc30a8a56d" />

### Compact table improvements

- Row click works from any cell (text, status icon, sparkline, location, etc.). The previous class-name whitelist (`euiTableCellContent` / `clickCellContent`) made most clicks silently no-op; now clicks open the flyout unless they originate from an interactive element (`a`, `button`, `[role="button"]`).
- Cursor is `pointer` on rows for affordance.
- Initial load now shows a loading spinner instead of "No monitors found"; the empty-state message only shows once the request resolves.

### Histogram fix (project monitors)

`useMonitorHistogramPerMonitor` and `useMonitorHistogramPerLocation` aggregated terms by `monitor.id` while documents are filtered and looked up by `config_id`. For project monitors `monitor.id` is `{projectId}-{name}` while `config_id` is a UUID, so aggregation buckets never matched and most rows rendered empty histograms. Aggregating by `config_id` aligns the keys for both project and UI-created monitors.

### Detail flyout refactor

**Location selector**
- Replaced the dropdown with a wrapping group of per-location buttons. Each button shows a status dot + label + status text (`EuiHealth`) so up/down/pending is visible at a glance.
- Renders synchronously from `monitor.locations` in `OverviewStatusMetaData` (mapping statuses through `useMonitorHealthColor`) — no extra API call, so the picker appears immediately.
- Wraps to multiple rows for monitors with many locations; small "Location" header sits above it.

**Status & duration**
- Replaced the per-location downtime histogram with the full `MonitorStatusPanel` heatmap (up/down/skipped, last 24h). `MonitorStatusPanel` now accepts optional `monitorId` and `locationLabel` props so it can be embedded without a route change; the underlying hook uses these overrides and skips the redundant `useSelectedMonitor`/`useSelectedLocation` fetches.
- Fixed empty histogram caused by reading the non-existent `monitor.locationId` — now uses the `locationId` prop passed to the flyout.
- Added an "All locations" toggle on the duration chart. When on, renders one duration line per location instead of the single-location + previous-period comparison.

**Layout & actions**
- Removed the standalone "Enabled (all locations)" switch from the flyout header; the equivalent toggle inside `MonitorDetailsPanel` is now visible (was previously hidden via `hideEnabled`).
- Added an "Edit monitor" button in the footer next to "Go to monitor".
- Flyout is wider and resizable (`size="m"`, `maxWidth={1000}`, `resizable`).

**Cleanup**
- Removed a duplicate `selectOverviewFlyoutConfig` definition in `state/overview/selectors.ts`.

## How to test

1. Open the Synthetics overview page.
2. Toggle **Group by config** and verify monitors are aggregated to one entry per config across locations.
3. Switch to the compact (table) view and click anywhere on a row (name, status icon, sparkline, location cell, last run, etc.) — the detail flyout should open every time.
4. Reload the page and confirm a loading spinner is shown briefly before rows render (and "No monitors found" only when the result is truly empty).
5. In the detail flyout:
   - The "Location" group should appear immediately with one button per location, each showing a status dot + label + status text.
   - Clicking a location button should switch the flyout to that location.
   - The status heatmap should populate for any monitor that has runs in the last 24h, **including project monitors** (previously empty for project monitors due to the histogram bug).
   - Toggle "All locations" on the duration chart and confirm one line per location is rendered.
   - The footer should show "Edit monitor" (links to the edit page) and "Go to monitor".
   - Drag the left edge to resize the flyout up to ~1000px.

## Risk

Scoped to the Synthetics overview (group-by-config + compact view) and the monitor detail flyout. The histogram aggregation key change is backward compatible because `config_id` is also used as the lookup key in the UI and has been indexed for the lifetime of the overview status pipeline.


<!--ONMERGE {"backportTargets":["8.19"]} ONMERGE-->

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

shahzad31 committed:
group by config

shahzad31 committed:
Merge branch 'main' of https://github.com/elastic/kibana into group-by-config

shahzad31 committed:
fix types

shahzad31 committed:
Merge branch 'main' of https://github.com/elastic/kibana into group-by-config

shahzad31 committed:
Merge branch 'main' of https://github.com/elastic/kibana into group-by-config

shahzad31 committed:
group by things

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/302018)
* Commit: 2503c8a1625b7ffb1d93db72b6d4cbdcd6aaa4d4

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/302018#0196fcf1-8f1f-427a-86f5-ae1723446780)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302018","number":302018,"commit":"2503c8a1625b7ffb1d93db72b6d4cbdcd6aaa4d4"}],"number":302018}
buildkite-pr-comment-->

shahzad31 committed:
Merge branch 'main' of https://github.com/elastic/kibana into group-by-config

shahzad31 committed:
update lint

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/302282)
* Commit: f4a916996c15c5afdb37a439ba8cc812aea70bb7

### Failed CI Steps
* [FTR Configs #44](https://buildkite.com/elastic/kibana-pull-request/builds/302282#01970bdb-e39f-4d33-9680-a2d8bf2a025c)
* [FTR Configs #56](https://buildkite.com/elastic/kibana-pull-request/builds/302282#01970bdb-e3a6-4284-b32e-1da0b161e17e)
* [FTR Configs #67](https://buildkite.com/elastic/kibana-pull-request/builds/302282#01970bdb-e3ac-4ed7-9577-b8f5244c0448)
* [FTR Configs #111](https://buildkite.com/elastic/kibana-pull-request/builds/302282#01970bdb-e3c5-4f5a-b893-71f82eb1c3ea)
* [FTR Configs #117](https://buildkite.com/elastic/kibana-pull-request/builds/302282#01970bdb-e3c8-4205-9f8b-53c73bc235db)
* [Synthetics @elastic/synthetics Tests](https://buildkite.com/elastic/kibana-pull-request/builds/302282#01970bc1-7728-48f0-a682-bfb44cffe52b)
* [Synthetics @elastic/synthetics Tests](https://buildkite.com/elastic/kibana-pull-request/builds/302282#01970be9-ac8f-4e20-bbfa-4834b536578a)

### History
* :broken_heart: [Build #302018](https://buildkite.com/elastic/kibana-pull-request/builds/302018) failed 2503c8a1625b7ffb1d93db72b6d4cbdcd6aaa4d4

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302282","number":302282,"commit":"f4a916996c15c5afdb37a439ba8cc812aea70bb7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302018","number":302018,"commit":"2503c8a1625b7ffb1d93db72b6d4cbdcd6aaa4d4"}],"number":302282}
buildkite-pr-comment-->

shahzad31 committed:
fix issues

shahzad31 committed:
update tests

shahzad31 committed:
update tests

shahzad31 committed:
update status

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/302382)
* Commit: 0e7ed3f8aa4a9c66b4cd0b2205b055f6d9d7696f

### Failed CI Steps
* [FTR Configs #56](https://buildkite.com/elastic/kibana-pull-request/builds/302382#01970cc3-1d35-4766-af7b-a82679aee6ce)
* [FTR Configs #67](https://buildkite.com/elastic/kibana-pull-request/builds/302382#01970cc3-1d3e-4423-85a7-b3bf583fb005)
* [FTR Configs #72](https://buildkite.com/elastic/kibana-pull-request/builds/302382#01970cc3-1d43-4102-ac86-a5f68dd3fa4d)
* [FTR Configs #86](https://buildkite.com/elastic/kibana-pull-request/builds/302382#01970cc3-1d4e-471f-91fc-ae4b11d60f87)
* [FTR Configs #89](https://buildkite.com/elastic/kibana-pull-request/builds/302382#01970cc3-1d50-46bf-982e-d0376dae186c)
* [FTR Configs #90](https://buildkite.com/elastic/kibana-pull-request/builds/302382#01970cc3-1d51-4b35-9663-2912dc37c7bb)
* [FTR Configs #117](https://buildkite.com/elastic/kibana-pull-request/builds/302382#01970cc3-1d60-42c2-9868-444cef29dcf2)
* [FTR Configs #123](https://buildkite.com/elastic/kibana-pull-request/builds/302382#01970cc3-1d62-43e0-b075-845a6e508550)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/302382#01970cc3-1d62-43e0-b075-845a6e508550) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/302382/jobs/01970cc3-1d62-43e0-b075-845a6e508550/artifacts/01970ce0-c8be-4201-9ce6-8e539b233b49) FTR Configs #<span></span>123 / Stateful Observability - Deployment-agnostic API integration tests SyntheticsAPITests inspectSyntheticsMonitor "before all" hook for "inspect http monitor"

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/e8fc261c36dd4decff2d3fd0d8594370363b8791) | [after](https://github.com/elastic/kibana/commit/0e7ed3f8aa4a9c66b4cd0b2205b055f6d9d7696f) | diff |
| --- | --- | --- | --- |
| `synthetics` | 1198 | 1205 | +7 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/e8fc261c36dd4decff2d3fd0d8594370363b8791) | [after](https://github.com/elastic/kibana/commit/0e7ed3f8aa4a9c66b4cd0b2205b055f6d9d7696f) | diff |
| --- | --- | --- | --- |
| `synthetics` | 969.2KB | 976.5KB | +7.2KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/e8fc261c36dd4decff2d3fd0d8594370363b8791) | [after](https://github.com/elastic/kibana/commit/0e7ed3f8aa4a9c66b4cd0b2205b055f6d9d7696f) | diff |
| --- | --- | --- | --- |
| `synthetics` | 27.8KB | 27.8KB | -4.0B |



### History
* :broken_heart: [Build #302282](https://buildkite.com/elastic/kibana-pull-request/builds/302282) failed f4a916996c15c5afdb37a439ba8cc812aea70bb7
* :broken_heart: [Build #302018](https://buildkite.com/elastic/kibana-pull-request/builds/302018) failed 2503c8a1625b7ffb1d93db72b6d4cbdcd6aaa4d4

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302382","number":302382,"commit":"0e7ed3f8aa4a9c66b4cd0b2205b055f6d9d7696f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302282","number":302282,"commit":"f4a916996c15c5afdb37a439ba8cc812aea70bb7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302018","number":302018,"commit":"2503c8a1625b7ffb1d93db72b6d4cbdcd6aaa4d4"}],"number":302382}
buildkite-pr-comment-->

cesco-f reviewed (dismissed):
I left some comments!

There are quite a few changes in this PR that seem like they could potentially break things. I’m still pretty new to the product, so I might not catch everything. It’s probably a good idea for someone else on the team with more experience to take a look as well, just to be safe.

Also, we should add some tests to cover these changes, just to make sure everything works as expected.

Last thing: the `Group by` `Location` does not work as expected.
![Screenshot 2025-05-28 at 10 55 12](https://github.com/user-attachments/assets/9fb2ded6-9ee6-49aa-8f90-4a6803cc4339)

shahzad31 committed:
Merge branch 'main' of https://github.com/elastic/kibana into group-by-config

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/321645)
* Commit: 7d07926dbb905f54b2d29a28799e795e3cca4f3e

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/321645#019830cb-ff4b-4bc6-9659-1bfde4fecf10)

### History
* :yellow_heart: [Build #302382](https://buildkite.com/elastic/kibana-pull-request/builds/302382) was flaky 0e7ed3f8aa4a9c66b4cd0b2205b055f6d9d7696f
* :broken_heart: [Build #302282](https://buildkite.com/elastic/kibana-pull-request/builds/302282) failed f4a916996c15c5afdb37a439ba8cc812aea70bb7
* :broken_heart: [Build #302018](https://buildkite.com/elastic/kibana-pull-request/builds/302018) failed 2503c8a1625b7ffb1d93db72b6d4cbdcd6aaa4d4

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321645","number":321645,"commit":"7d07926dbb905f54b2d29a28799e795e3cca4f3e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302382","number":302382,"commit":"0e7ed3f8aa4a9c66b4cd0b2205b055f6d9d7696f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302282","number":302282,"commit":"f4a916996c15c5afdb37a439ba8cc812aea70bb7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302018","number":302018,"commit":"2503c8a1625b7ffb1d93db72b6d4cbdcd6aaa4d4"}],"number":321645}
buildkite-pr-comment-->

shahzad31 committed:
wip

shahzad31 committed:
update tests

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/322335)
* Commit: f74aca0bcf6ea42c5e4aa99a9913bb8b820cc67d

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/99987fe49943644168b2a56ef4fd2b0805c9c85d) | [after](https://github.com/elastic/kibana/commit/f74aca0bcf6ea42c5e4aa99a9913bb8b820cc67d) | diff |
| --- | --- | --- | --- |
| `synthetics` | 1367 | 1372 | +5 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/99987fe49943644168b2a56ef4fd2b0805c9c85d) | [after](https://github.com/elastic/kibana/commit/f74aca0bcf6ea42c5e4aa99a9913bb8b820cc67d) | diff |
| --- | --- | --- | --- |
| `synthetics` | 1.0MB | 1.0MB | +6.1KB |



### History
* :broken_heart: [Build #321645](https://buildkite.com/elastic/kibana-pull-request/builds/321645) failed 7d07926dbb905f54b2d29a28799e795e3cca4f3e
* :yellow_heart: [Build #302382](https://buildkite.com/elastic/kibana-pull-request/builds/302382) was flaky 0e7ed3f8aa4a9c66b4cd0b2205b055f6d9d7696f
* :broken_heart: [Build #302282](https://buildkite.com/elastic/kibana-pull-request/builds/302282) failed f4a916996c15c5afdb37a439ba8cc812aea70bb7
* :broken_heart: [Build #302018](https://buildkite.com/elastic/kibana-pull-request/builds/302018) failed 2503c8a1625b7ffb1d93db72b6d4cbdcd6aaa4d4

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/322335","number":322335,"commit":"f74aca0bcf6ea42c5e4aa99a9913bb8b820cc67d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321645","number":321645,"commit":"7d07926dbb905f54b2d29a28799e795e3cca4f3e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302382","number":302382,"commit":"0e7ed3f8aa4a9c66b4cd0b2205b055f6d9d7696f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302282","number":302282,"commit":"f4a916996c15c5afdb37a439ba8cc812aea70bb7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302018","number":302018,"commit":"2503c8a1625b7ffb1d93db72b6d4cbdcd6aaa4d4"}],"number":322335}
buildkite-pr-comment-->

shahzad31 committed:
Merge remote-tracking branch 'upstream/main' into group-by-config

Made-with: Cursor

# Conflicts:
#	x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/common/components/monitor_status.tsx
#	x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/monitors_page/hooks/use_can_use_public_loc_id.ts
#	x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/monitors_page/hooks/use_monitor_histogram.ts
#	x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/monitors_page/management/monitor_list_table/monitor_locations.tsx
#	x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/monitors_page/overview/overview/actions_popover.tsx
#	x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/monitors_page/overview/overview/compact_view/components/monitors_duration.tsx
#	x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/monitors_page/overview/overview/metric_item/metric_item.tsx
#	x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/monitors_page/overview/overview/metric_item/metric_item_body.tsx
#	x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/monitors_page/overview/overview/overview_grid.tsx
#	x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/hooks/use_status_by_location.tsx
#	x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/state/overview_status/selectors.ts
#	x-pack/solutions/observability/plugins/synthetics/server/routes/overview_status/overview_status_service.ts

shahzad31 committed:
Fix bugs and address review feedback

- Fix critical bug: pass location id (not label) to getConfigStatusByLocation in use_filtered_group_monitors
- Add empty-array guards for monitor.locations[0] access across 12 files
- Refactor use_metric_subtitle to reduce repetitive status filtering code
- Fix inconsistent group_fields value 'location.id' to match state field 'locationId'
- Clean up redundant locationId: locationId in actions_popover

Made-with: Cursor

shahzad31 committed:
Add unit tests for grouping logic

- selectors.test.ts: tests for getStatusByConfig, selectOverviewStatus,
  and separateByLocation (per-location splitting for non-monitor group-by)
- use_status_by_location_overview.test.ts: tests for getConfigStatusByLocation
  covering configId, composite keys, multi-location, and edge cases
- use_metric_subtitle.test.tsx: tests for subtitle generation with various
  location statuses (down, up, pending, mixed, empty)

Made-with: Cursor

shahzad31 committed:
Add server-side unit tests for grouping logic in processOverviewStatus

Tests cover multi-location grouping under configId, overallStatus
precedence (down > up > pending), pending-to-down/up promotion with
sorting, disabled monitor grouping, and timestamp selection.

Made-with: Cursor

shahzad31 committed:
Fix grouping bugs, optimize selectors, and fix overview_trends request

P0: Fix location grouping using wrong field keys in grid_items_by_group
P0: Fix up/down counter double-counting in processOverviewStatus
P0: Fix overview_trends 400 error (locationId -> locations array)
P1: Memoize selectOverviewStatus with createSelector
P1: Fix useFilteredGroupMonitors to check overallStatus for multi-loc
P2: Remove redundant useOverviewStatus call in OverviewCardView
P2: Add missing React key on group map fragments
Made-with: Cursor

shahzad31 committed:
Fix infinite overview_trends requests caused by trendData dependency loop

Use a ref to read trendData without including it in the useEffect
dependency array, breaking the dispatch -> state update -> re-dispatch
cycle.

Made-with: Cursor

shahzad31 committed:
Fix infinite re-render loops causing 100% CPU on overview page

- useMetricSubtitle: Add missing dependency array to useEffect that
  dispatches getServiceLocations, preventing it from firing every render
- OverviewCardView: Guard setRowCount with a ref check to prevent
  state updates during render when the value hasn't changed

Made-with: Cursor

shahzad31 committed:
Fix infinite re-render loops and improve overview performance

- Create targeted selectors (selectOverviewGroupBy, selectOverviewView, etc.)
  to prevent unnecessary re-renders from trendStats changes
- Memoize arrays in useInfiniteOverviewTrendsRequests and GroupGridCardContent
  to break re-render -> new array -> effect -> dispatch cycles
- Fix trendStatsBatch.get reducer to mark configId+locationId cache keys
- Extract UnGroupedCardView to isolate selectOverviewTrends subscription
- Default group-by to 'monitor' and persist selection in localStorage
- Fix LocationsBadge layout in metric item cards (grow={false})

Made-with: Cursor

shahzad31 committed:
[Synthetics] Improve flyout UX and compact table responsiveness

- Progressive loading in monitor detail flyout: show header, duration
  chart, and downtime history immediately using available data, with
  inline spinners for detailed sections still loading
- Replace useOverviewStatus() with useSelector(selectOverviewStatus)
  in flyout to eliminate redundant API call on open
- Add downtime history chart (MonitorBarSeries) to the flyout body
- Quiet fetch no longer sets loading=true, preventing table spinner
  during background refreshes
- Hide URL, Tags, and Downtime History columns when flyout is open
  to give more space to the remaining columns
- Remove Duration column, add frequency badge (@every Xm) to Tags
- Fix table layout to respect column widths (tableLayout="fixed")
- Add selectOverviewFlyoutConfig selector

Made-with: Cursor

shahzad31 committed:
Merge branch 'synthetics-flyout-ui-improvements' into group-by-config

Made-with: Cursor

# Conflicts:
#	x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/monitors_page/overview/overview/compact_view/hooks/use_monitors_table_columns.tsx

shahzad31 committed:
[Synthetics] Fix histogram aggregation key for project monitors

Histogram hooks aggregate by `monitor.id` while documents are filtered
and looked up by `config_id`. For project monitors `monitor.id` is
`{projectId}-{name}` while `config_id` is a UUID, so aggregation buckets
never matched the UI lookup keys and most rows rendered empty histograms.

Switch the terms aggregation field to `config_id` so keys align for both
project and UI-created monitors.

Made-with: Cursor

shahzad31 committed:
[Synthetics] Improve compact table row click and loading UX

- Make row click reliable from any cell. Previously a class-name
  whitelist (`euiTableCellContent` / `clickCellContent`) caused most
  clicks on cell content (icons, charts, sparklines, plain text) to be
  ignored. Now we open the flyout unless the click originates from an
  interactive element (`a`, `button`, `[role="button"]`).
- Add `cursor: pointer` on rows for affordance.
- Drop the brittle `clickCellContent` class and inline `onClick` from
  the name column; the row-level handler covers it.
- Replace the bare "No items found" placeholder shown during initial
  load with a loading spinner; only show the empty-state copy once the
  request has resolved.

Made-with: Cursor

shahzad31 committed:
[Synthetics] Refactor monitor detail flyout

Streamlines the detail flyout to make per-location triage faster and
gives the flyout more room.

Location selector
- Replace the dropdown with a wrapping group of per-location buttons.
  Each button shows a status dot + label + status text via EuiHealth, so
  the up/down/pending state is visible at a glance and users can switch
  locations in one click.
- Render synchronously from `monitor.locations` in
  `OverviewStatusMetaData` (mapping statuses through
  `useMonitorHealthColor`) instead of waiting on `useStatusByLocation`,
  so the picker appears immediately.
- The button group wraps to multiple rows for monitors with many
  locations; a small "Location" label sits above it.

Status & duration
- Replace the per-location downtime histogram with the full
  `MonitorStatusPanel` heatmap (up/down/skipped, last 24h). To support
  embedding without a route change, `MonitorStatusPanel` now accepts
  optional `monitorId` and `locationLabel` props and the underlying hook
  uses these overrides (and skips the redundant
  `useSelectedMonitor`/`useSelectedLocation` refetches).
- Fix empty heatmap caused by reading `monitor.locationId` (which
  doesn't exist) — use the `locationId` prop already passed to the
  flyout.
- Add an "All locations" toggle on the duration chart. When on, render
  one duration line per location instead of the single-location +
  previous-period comparison.

Layout & actions
- Remove the standalone "Enabled (all locations)" switch from the
  header; the equivalent toggle inside `MonitorDetailsPanel` is now
  visible (was previously hidden via `hideEnabled`).
- Add an "Edit monitor" button in the footer next to "Go to monitor".
- Make the flyout wider and resizable (`size="m"`, `maxWidth={1000}`,
  `resizable`).

Cleanup
- Remove duplicate `selectOverviewFlyoutConfig` definition in
  `state/overview/selectors.ts`.

Made-with: Cursor

shahzad31 committed:
[Synthetics] Fix overview status & trends bugs from PR review

- `getStatusByConfig` previously only checked `downConfigs` and
  `upConfigs`. A disabled or always-pending monitor was reported as
  PENDING regardless of its actual stored status. Look up the config
  across all four buckets (down/up/pending/disabled) for both the bare
  configId key and the `${configId}-${locationId}` composite key. Tests
  added for the new behaviour.
- `OverviewStatusService.getStatusByLocation` was wrapping the URL value
  in `String(...)` before checking it, producing the literal string
  `"undefined"` for monitors without `url.full.keyword`. Use the raw
  value, null-check it, and only stringify when defined. Inline
  snapshots updated to reflect the corrected output.
- `useOverviewTrendsRequests` and `SingleMonitorView` checked the trend
  cache using only `locations[0].id`. Once the first location resolved,
  no further request was issued, so additional locations never got
  trend data. Now we look at every location and only fetch the missing
  ones.

Made-with: Cursor

shahzad31 committed:
[Synthetics] Fix histogram filter typing and locations badge hover

- The histogram query hooks cast the `bool.filter` array to a single
  `estypes.QueryDslQueryContainer`. It compiles today but breaks any
  consumer that calls array methods on it. Cast to
  `QueryDslQueryContainer[]`.
- `LocationsBadge` closed its popover on the trigger's `mouseout`/`blur`
  events, but the popover menu is rendered in a portal outside the
  trigger. Moving the cursor toward the menu fired `mouseout` and
  dismissed the popover before users could click an item. Drop those
  handlers and rely on `EuiPopover`'s built-in outside-click / Escape
  handling via `closePopover`.

Made-with: Cursor

shahzad31 committed:
[Synthetics] Fix sort-by-status with multi-location monitors

When `Group by` is not set to `monitor`, multi-location monitors are
split into one entry per location keyed `${configId}-${locationId}`. The
splits used to inherit the *parent* monitor's status bucket (e.g. all
locations of a down monitor stayed in `downConfigs` even when individual
locations were pending). That broke the sort-by-status concat in
`useMonitorsSortedByStatus` — pending splits would render alongside
their down sibling instead of being grouped at the bottom.

Now multi-location splits are routed to the bucket matching their
per-location status (`up` → `upConfigs`, `pending` → `pendingConfigs`,
etc.), while single-location entries keep their server-assigned bucket.

Adds a regression test demonstrating that a down monitor with mixed
location statuses redistributes correctly across `downConfigs` and
`pendingConfigs`.

Made-with: Cursor

shahzad31 committed:
[Synthetics] Set default width for Spaces column in compact table

The Spaces column had no `width` set so the table layout (`fixed`)
allocated it the remaining space, leaving large empty gaps when each
row only renders a single space avatar. Use `120px` to match the
Locations column for visual consistency while still leaving room for
two or three avatars when a monitor is shared across spaces.

Made-with: Cursor

shahzad31 committed:
type fixes

shahzad31 committed:
[Synthetics] Overview perf wins, display options, regression tests

Performance:
- Fix SyntheticsRefreshContextProvider double-bumping `lastRefresh` on
  mount, which was fan-fanning out a duplicate fetch to every consumer
  hook (filters, overview status, histograms, trends).
- Drop the 100ms initial-mount `useDebounce` in `useOverviewStatus` and
  `useMonitorList` — the saga's existing `debounce(300)` already
  rate-limits pageState changes; the client-side delay was pure latency.
- Skip histogram + trend fetches in compact view while the flyout is
  open (the columns they feed are hidden in that layout).
- Stabilise the cache key in `useMonitorHistogramPerLocation/PerMonitor`
  via `useMemo` so we stop running JSON.stringify on every render.
- Gate `refreshOverviewTrends.get()` on `view === 'cardView'` and
  `loaded` so compact view doesn't fire a duplicate trend refresh on
  every auto-refresh tick.
- Use `space?.id` (string) instead of the whole `space` object in the
  compact-table columns memo to keep cell identity stable across the
  active-space resolution.

Display options + URL sync:
- New `useOverviewDisplayOptions` hook backed by a module-level store
  via `useSyncExternalStore` so toggling "Absolute timestamps" applies
  immediately to all consumers, including across tabs.
- New `DisplayOptionsPopover` consolidating cosmetic + filter toggles.
- New `useSyncDateRangeFilter` mounted at page level to keep the URL
  the source of truth for the date-range filter even when `OverviewGrid`
  unmounts behind a `<NoMonitorsFound />` empty state.

Regression tests:
- `use_sync_date_range_filter.test.tsx` covering URL → pageState sync
  and conditional loud-fetch behaviour.
- `use_overview_display_options.test.ts` covering cross-instance
  propagation, persistence, and resetOptions.

Made-with: Cursor

shahzad31 committed:
[Synthetics] Move overview perf wins to a separate PR

Reverts the three perf-only hunks (`synthetics_refresh_context.tsx`
double-bump fix and the dropped initial-mount `useDebounce` from
`useOverviewStatus`/`useMonitorList`) so they can be reviewed and
landed independently of the group-by-config + compact-view work.

These changes now live in their own PR off `main`. No behaviour change
in this commit relative to current `main` for the three reverted files.

Made-with: Cursor

kibanamachine committed:
Changes from node scripts/lint.js --fix

elasticmachine commented:
Pinging @elastic/actionable-obs-team (Team:actionable-obs)

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

miguelmartin-elastic commented on x-pack/solutions/observability/plugins/synthetics/server/routes/overview_status/overview_status_service.ts:
```diff
@@ -278,37 +412,87 @@ export class OverviewStatusService {
+          upConfigs[meta.configId] ||
+          pendingConfigs[meta.configId]
+        ) {
+          const existingMeta =
+            downConfigs[meta.configId] || upConfigs[meta.configId] || pendingConfigs[meta.configId];
+          existingMeta.locations.push(location);
+          // check if urls is missing from existing meta and update it
+          if (!existingMeta.urls && meta.urls) {
+            existingMeta.urls = meta.urls;
+          }
+          // also update timestamp if it is missing or older
+          if (
+            !existingMeta.timestamp ||
+            (meta.timestamp && moment(meta.timestamp).isAfter(existingMeta.timestamp))
+          ) {
+            existingMeta.timestamp = meta.timestamp;
+          }
+          if (status === MONITOR_STATUS_ENUM.DOWN) {
+            existingMeta.overallStatus = MONITOR_STATUS_ENUM.DOWN;
```
`processOverviewStatus` decides the bucket from the first location processed and  
  only mutates `overallStatus` on later DOWN locations — never moves the entry
  between buckets. So the bucket can disagree with `overallStatus` depending on ES  
  iteration order.
                                                                                    
  Example — monitor M with locations L1 (UP) and L2 (DOWN):                         
    • DOWN-first order: L2 → downConfigs[M]; L1 → push, overallStatus stays 'down'
  ✅                                                                                
    • UP-first order:   L1 → upConfigs[M];   L2 → push, overallStatus mutated to
  'down' ❌                                                                         
                         → M stays in upConfigs with overallStatus: 'down'

miguelmartin-elastic commented on x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/monitors_page/overview/overview/monitor_detail_flyout.tsx:
```diff
@@ -222,19 +244,34 @@ export function LoadingState() {
     const allConfigs = Object.values({
-      ...(overviewStatus?.upConfigs ?? {}),
-      ...(overviewStatus?.downConfigs ?? {}),
+      ...(overviewStatus.upConfigs ?? {}),
+      ...(overviewStatus.downConfigs ?? {}),
```
Apparently this is no new behaviour but, when monitor is in PENDING status, is not retrieved here, and in the flyout no Location Badge is shown.  I didn't see any "broken" component per sé, but wanted to duble check with you that this is expected behaviour

miguelmartin-elastic commented on x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/monitors_page/management/monitor_list_table/monitor_locations.tsx:
```diff
@@ -5,46 +5,59 @@
+    });
+
+    return (
+      <LocationStatusBadges configId={configId} locations={locationsToDisplay} loading={false} />
```
Clicking on the LocationStatusBadge within the row redirects the user to the Monitor Detail Page. Not sure if that's the intended behaviour

miguelmartin-elastic commented on x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/monitors_page/overview/overview/metric_item/locations_badge.tsx:
```diff
@@ -0,0 +1,114 @@
+          key: location.id,
+          name: location.label,
+          icon: <EuiIcon type="dot" color={getColor(location.status)} />,
+          onClick: () => {
```
Let's say I have one Monitor in LocA and LocB. 

When I hover on LocationsBadge and click on LocB, the flyout opens but always the first location is selected: LocA, not the location I clicked on

miguelmartin-elastic reviewed (commented):
There are some comments from Macroscope I think are worthwhile to check. I also added a blocking one. Rest of my comments may be just me misunderstanding the expected behaviour. 

Anyway, this is a massive PR so I'll take another look tomorrow with a fresh mind. 

Also, we should create a tech debt issue to refactor this in the feature, wdyt @shahzad31 ?

shahzad31 committed:
Address PR review feedback: fix bucket misplacement, flyout bugs, and perf

- Fix processOverviewStatus so monitors with mixed UP+DOWN locations
  always land in downConfigs regardless of ES iteration order
- Include pendingConfigs and disabledConfigs in flyout's allConfigs
  lookup so location badges render for all monitor states
- Wire LocationsBadge onClick to open flyout for the clicked location
  instead of always defaulting to the first location
- Fix absoluteTimestamps tooltip inversion in monitor_status_col so
  bold text matches the user's preference
- Wrap histogramsById computation in useMemo in both histogram hooks
  to prevent unnecessary downstream re-renders

Made-with: Cursor

shahzad31 committed:
Merge branch 'main' of https://github.com/elastic/kibana into group-by-config

Made-with: Cursor

# Conflicts:
#	x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/monitors_page/management/monitor_list_table/columns.tsx
#	x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/monitors_page/management/monitor_list_table/monitor_locations.tsx

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/434621)
* Commit: 7a7edc6c32ff7d4595f92d2a31e16c3ccf9b67d3

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/434621#019dd2ae-7f69-435a-a505-99d036464b79)

### History
* :broken_heart: [Build #434326](https://buildkite.com/elastic/kibana-pull-request/builds/434326) failed 4e366d0abcd1f66b7c399ada06b82e3b9cfedfce
* :broken_heart: [Build #434299](https://buildkite.com/elastic/kibana-pull-request/builds/434299) failed 74823861feb2f237c2728f2f84cb145dcdaa7364

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/434621","number":434621,"commit":"7a7edc6c32ff7d4595f92d2a31e16c3ccf9b67d3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/434326","number":434326,"commit":"4e366d0abcd1f66b7c399ada06b82e3b9cfedfce"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/434299","number":434299,"commit":"74823861feb2f237c2728f2f84cb145dcdaa7364"}],"number":434621}
buildkite-pr-comment-->

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

shahzad31 committed:
fix i18n

shahzad31 committed:
Fix react-hooks/exhaustive-deps lint errors in histogram hooks

Move byIdsBuckets inside useMemo to fix exhaustive-deps warnings.

Made-with: Cursor

miguelmartin-elastic commented on x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/monitors_page/overview/overview/metric_item/metric_item.tsx:
```diff
@@ -242,7 +242,20 @@ export const MetricItem = ({
+                    <MetricItemBody
+                      monitor={monitor}
+                      onLocationClick={(locId, locLabel) => {
+                        onClick({
```
Even passing `onLocationClick` the flyout is opening with the first location selected. I think it's because there is a `onElementClick` in this file in line ~`217`

miguelmartin-elastic commented on x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/monitors_page/common/display_options_popover.tsx:
```diff
@@ -0,0 +1,191 @@
+ *
+ * - **Cosmetic** (e.g. absolute timestamps): saved to `localStorage` via
+ *   `useOverviewDisplayOptions`. Per-user, per-space, never in the URL.
+ * - **Semantic** (e.g. `filterByDateRange`): URL-driven so the view is
```
If the dateRange switch is off the date picker has no effect in the list. This feels kinda confusing for users at first if they don't know this switch exists. I'd suggest 

A) Remove the switch and make the date picker always relevant, or
B) Add a tooltip or something in the date picker so the user is aware of this switch

shahzad31 committed:
fix on location click

shahzad31 committed:
remove date range filter changes from group-by-config

These changes are being extracted into a separate PR so
group-by-config can be merged independently.

Made-with: Cursor

shahzad31 committed:
remove date picker from overview page header

Missed in the previous date-range removal — restores the shared
RefreshButton + CreateMonitorButton header for both overview and
management routes.

Made-with: Cursor

shahzad31 committed:
Fix Monitor Detail Flyout Jest tests: add charts service mock

The refactored flyout now renders MonitorStatusPanel which requires
the charts Kibana service. Add chartPluginMock to the test helper's
mock core and update the status assertion to match the new flyout
layout (heatmap caption instead of removed Pending badge).

Made-with: Cursor

miguelmartin-elastic reviewed (approved):
Functionally lgtm! 

FTR: we'll tackle refactoring as tech debt

shahzad31 committed:
Address PR review feedback: fix badge throttle bypass and CodeQL finding

- Remove redundant onMouseOver on EuiBadge that bypassed the throttled
  popover opener on the parent div
- Add maxSize to inner locationIds schema.arrayOf to satisfy CodeQL
  unbounded-array validation

Made-with: Cursor

shahzad31 committed:
Fix loading state for pending monitors, adapt status counts to group-by, default grouping to none, and add group-by tour

- Fix trendStatsBatch reducer to clear configId+locationId composite keys
  so pending monitors no longer show perpetual loading spinners
- Recompute up/down/pending/disabled counts in selectOverviewStatus to
  reflect the active group-by selection
- Change default grouping from 'monitor' to 'none' to match main branch
- Add EuiTourStep highlighting the new "Group by monitor" option

Made-with: Cursor

kibanamachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/435200)
* Commit: 62584bf03d62138dc05aac323899095370c0ec5e

### Failed CI Steps
* [FTR Configs #67](https://buildkite.com/elastic/kibana-pull-request/builds/435200#019dd57b-f8dd-46a8-866a-7bb29a3adce6)
* [FTR Configs #80](https://buildkite.com/elastic/kibana-pull-request/builds/435200#019dd57b-f8eb-433e-af36-1f1790dfc5e7)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/435200#019dd57b-f8eb-433e-af36-1f1790dfc5e7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/435200/jobs/019dd57b-f8eb-433e-af36-1f1790dfc5e7/artifacts/019dd597-5d14-4eda-8e88-5cda5d790263) FTR Configs #<span></span>80 / Agent Builder agents Edit agent should edit agent name

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/f5b6de7e48997c654182db84fceca6ffad8d1032) | [after](https://github.com/elastic/kibana/commit/62584bf03d62138dc05aac323899095370c0ec5e) | diff |
| --- | --- | --- | --- |
| `synthetics` | 1288 | 1294 | +6 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f5b6de7e48997c654182db84fceca6ffad8d1032) | [after](https://github.com/elastic/kibana/commit/62584bf03d62138dc05aac323899095370c0ec5e) | diff |
| --- | --- | --- | --- |
| `synthetics` | 1.1MB | 1.1MB | +17.1KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/f5b6de7e48997c654182db84fceca6ffad8d1032) | [after](https://github.com/elastic/kibana/commit/62584bf03d62138dc05aac323899095370c0ec5e) | diff |
| --- | --- | --- | --- |
| `synthetics` | 29.3KB | 29.3KB | +1.0B |

<details><summary>Unknown metric groups</summary>

#### ESLint disabled in files
| id | [before](https://github.com/elastic/kibana/commit/f5b6de7e48997c654182db84fceca6ffad8d1032) | [after](https://github.com/elastic/kibana/commit/62584bf03d62138dc05aac323899095370c0ec5e) | diff |
| --- | --- | --- | --- |
| `synthetics` | 5 | 6 | +1 |

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/f5b6de7e48997c654182db84fceca6ffad8d1032) | [after](https://github.com/elastic/kibana/commit/62584bf03d62138dc05aac323899095370c0ec5e) | diff |
| --- | --- | --- | --- |
| `synthetics` | 60 | 59 | -1 |


</details>


### History
* :yellow_heart: [Build #434859](https://buildkite.com/elastic/kibana-pull-request/builds/434859) was flaky 43f0e8247e288f141d3796219a9b76003a258b50
* :green_heart: [Build #434780](https://buildkite.com/elastic/kibana-pull-request/builds/434780) succeeded 664ecd719db8627e3f23d9feb5e8b1a9fb6e2fd1
* :broken_heart: [Build #434655](https://buildkite.com/elastic/kibana-pull-request/builds/434655) failed ec5582646f6b3009be99432fb5797ec9762ebc8c
* :broken_heart: [Build #434629](https://buildkite.com/elastic/kibana-pull-request/builds/434629) failed ea19f86bcbca521aceca203b5e35ed243a492377
* :broken_heart: [Build #434621](https://buildkite.com/elastic/kibana-pull-request/builds/434621) failed 7a7edc6c32ff7d4595f92d2a31e16c3ccf9b67d3
* :broken_heart: [Build #434326](https://buildkite.com/elastic/kibana-pull-request/builds/434326) failed 4e366d0abcd1f66b7c399ada06b82e3b9cfedfce

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/435200","number":435200,"commit":"62584bf03d62138dc05aac323899095370c0ec5e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/434859","number":434859,"commit":"43f0e8247e288f141d3796219a9b76003a258b50"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/434780","number":434780,"commit":"664ecd719db8627e3f23d9feb5e8b1a9fb6e2fd1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/434655","number":434655,"commit":"ec5582646f6b3009be99432fb5797ec9762ebc8c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/434629","number":434629,"commit":"ea19f86bcbca521aceca203b5e35ed243a492377"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/434621","number":434621,"commit":"7a7edc6c32ff7d4595f92d2a31e16c3ccf9b67d3"}],"number":435200}
buildkite-pr-comment-->

kibanamachine commented:
Starting backport for target branches: 8.19

https://github.com/elastic/kibana/actions/runs/25094048708

kibanamachine commented:
## 💔 All backports failed

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.19|**Backport failed because of merge conflicts**<br><br>You might need to backport the following PRs to 8.19:<br> - [[Synthetics] Fix cross-space monitor detail navigation in management list!! (#265647)](https://github.com/elastic/kibana/pull/265647)|

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 220082
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 220082` locally
cc: @shahzad31

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 220082` locally
cc: @shahzad31
